app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :signed_in_user, only: %i[new create]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user.id
+    @post.save
+    redirect_to root_path
```

Comment:
Even if it's working, I suggest you use **_url**
https://www.viget.com/articles/rails-named-routes-path-vs-url/

Reply:
Fixed it